plugin/evm/validators/state.go

```diff
@@ -293,3 +339,14 @@ func (v *validatorData) getLastUpdated() time.Time {
 func (v *validatorData) getStartTime() time.Time {
 	return time.Unix(int64(v.StartTime), 0)
 }
+
+// constantsAreUnmodified returns true if the constants of this validator have
+// not been modified compared to the other validator.
+func (v *validatorData) constantsAreUnmodified(o interfaces.Validator) bool {
+	if v.validationID != o.ValidationID {
```

Comment:
Is there ever a case where we would expect to call this with a different validation ID? Seems likely that would indicate a bug itself and shouldn't be allowed.

Reply:
It's not possible to update a validator with a different vID. It should return a not found. That function is more like `newOrCanBeUpdated"` but we only use it in update. so by just looking at that function it looks like we should not allow this. changed it.